File: lib/std/core/game.lua
require 'std.core.util'

local hook = require 'std.core.hook'

game = game or {}

function game.getElapsedTime()
    return TimerGetElapsed(game.__tracker)
end

function game.isSingleplayer()
    local count = 0
    for _, v in pairs(player.getAll()) do
        if v.isHuman and v.isPlaying then
            count = count + 1
        end
    end
    
    return count == 1
end

hook.once(function()
    local tracker = CreateTimer()
    TimerStart(tracker, 60 * 60 * 24 * 365, false, nil)
    game.__tracker = tracker
end)

File: ts/src/tsstd/core/game.ts
import Player from "./types/player"

let context = util.contextFn("game", () => {
    let tracker = CreateTimer()
    TimerStart(tracker, 60 * 60 * 24 * 365, false, undefined)
    return {
        tracker: tracker
    }
})

export function elapsedTime(): number {
    return TimerGetElapsed(context.tracker)
}

export function isSinglePlayer(): boolean {
    let count = 0
    for (let player of Player.all) {
        if (player.isHuman && player.isPlaying) {
            count++
        }
    }
    return count == 1
}
